modules/channel-smooch/src/backend/client.ts

```diff
@@ -69,28 +69,34 @@ export class SmoochClient {
     }
 
     for (const message of payload.messages) {
-      if (message.type !== 'text') {
-        continue
+      if (this.config.forwardRawPayloads.includes(`smooch-${message.type}`)) {
+        await this.receiveMessage(`smooch-${message.type}`, {}, message)
       }
 
-      await this.bp.events.sendEvent(
-        this.bp.IO.Event({
-          botId: this.botId,
-          channel: 'smooch',
-          direction: 'incoming',
-          type: 'text',
-          payload: {
-            type: 'text',
-            text: message.text
-          },
-          preview: message.text,
-          threadId: payload.conversation._id,
-          target: payload.appUser._id
-        })
-      )
+      if (message.type === 'text') {
+        await this.receiveMessage('text', { type: 'text', text: message.text }, message)
+      }
     }
   }
 
+  async receiveMessage(type: string, payload: any, rawMessage: Message) {
+    const rawPayload = this.config.forwardRawPayloads.includes(type)
+      ? { channel: { smooch: { message: rawMessage } } }
+      : {}
+
+    await this.bp.events.sendEvent(
+      this.bp.IO.Event({
+        botId: this.botId,
+        channel: 'smooch',
+        direction: 'incoming',
+        type,
+        payload: { ...rawPayload, payload },
+        threadId: payload.conversation._id,
+        target: payload.appUser._id
```

Comment:
You don't seem to be passing these values (`payload.conversation._id` and `payload.appUser._id`) when calling `receiveMessage`. Am I missing something here?

Reply:
Yes this fixed